### ISSUE TITLE
Pub publish is including build folder

### ISSUE THREAD

**Issue by edsulaiman** [dart-lang/pub#3109]
## Problem
On the previous version of flutter (stable 2.3.0) publishing a package will not including the build folder. But, after upgrading flutter to 2.5.0. Publishing a package using `dart pub publish` is including a build folder or ignoring the .gitignore file.

``` shell
Do you want to publish google_ui 2.5.0 (y/N)? y
Uploading...
Uncompressed package archive is too large (size > 104857600).
```
after I run `flutter clean` on the example folder publishing is succeed 
``` shell
Do you want to publish google_ui 2.5.0 (y/N)? y
Uploading...
Successfully uploaded package.
```

## Steps to Reproduce
- Clone [google_ui](https://github.com/edsulaiman/google_ui)
- Run `flutter run` to build and debug example
- Run `dart pub publish --dry-run`
- Run `dart pub publish`

**Comment by edsulaiman** [comment]
I downloaded [my package archive ](https://pub.dev/packages/google_ui/versions) and comparing them

google_ui 2.4.1 published using flutter 2.3.0
![image](https://user-images.githubusercontent.com/52323183/133231720-f34e6dae-d0f5-4314-929b-751e96ca1275.png)

google_ui 2.5.0 published using flutter 2.5.0
![image](https://user-images.githubusercontent.com/52323183/133231772-b25e4909-762f-41d7-bb13-7742b8bd6670.png)

As you can see in the latest version of my package (google_ui 2.5.0) the .gitignore file is not included in the published archive

**Comment by sigurdm** [comment]
This is a duplicate of: https://github.com/dart-lang/pub/issues/3102
For now the workaround is to place a .gitignore/.pubignore in the project itself or publishing with an older pub.
